src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -79,36 +80,41 @@ public void discoverClusterNodesAndSlots(Jedis jedis) {
   }
 
   public void discoverClusterSlots(Jedis jedis) {
-    w.lock();
+    //If rediscovering is already in process - no need to start one more same rediscovering, just return
+    if (!rediscovering) {
+      w.lock();
+      rediscovering = true;
```

Comment:
I guess it makes sense. Is there a cleaner way to do this instead of repeating the condition twice?